src/main/java/de/numcodex/feasibility_gui_backend/query/broker/direct/DirectBrokerClient.java

```diff
@@ -1,229 +1,215 @@
-    }
+  @Override
+  public String createQuery(Long backendQueryId) {
+    var brokerQuery = DirectQuery.create();
+    var brokerQueryId = brokerQuery.getQueryId();
+    brokerQueries.put(brokerQueryId, brokerQuery);
+    brokerToBackendQueryIdMapping.put(brokerQueryId, backendQueryId);
 
-    @Override
-    public BrokerClientType getBrokerType() {
-        return DIRECT;
-    }
+    return brokerQueryId;
+  }
 
-    @Override
-    public void addQueryStatusListener(QueryStatusListener queryStatusListener) {
-        listeners.add(queryStatusListener);
-    }
+  @Override
+  public void addQueryDefinition(String brokerQueryId, String mediaType, String content) throws QueryNotFoundException {
+    findQuery(brokerQueryId).addQueryDefinition(mediaType, content);
+  }
 
-    @Override
-    public String createQuery(Long backendQueryId) {
-        var brokerQuery = DirectQuery.create();
-        var brokerQueryId = brokerQuery.getQueryId();
-        brokerQueries.put(brokerQueryId, brokerQuery);
-        brokerToBackendQueryIdMapping.put(brokerQueryId, backendQueryId);
+  @Override
+  public abstract void publishQuery(String brokerQueryId) throws QueryNotFoundException, IOException;
 
-        return brokerQueryId;
+  @Override
+  public void closeQuery(String brokerQueryId) throws QueryNotFoundException {
+    if (brokerQueries.remove(brokerQueryId) == null) {
+      throw new QueryNotFoundException(brokerQueryId);
     }
-
-    @Override
-    public void addQueryDefinition(String brokerQueryId, String mediaType, String content) throws QueryNotFoundException {
-        findQuery(brokerQueryId).addQueryDefinition(mediaType, content);
+  }
+
+  @Override
+  public int getResultFeasibility(String brokerQueryId, String siteId) throws QueryNotFoundException, SiteNotFoundException {
+    return findQuery(brokerQueryId).getSiteResult(siteId);
+  }
+
+  @Override
+  public List<String> getResultSiteIds(String brokerQueryId) throws QueryNotFoundException {
+    return findQuery(brokerQueryId).getSiteIdsWithResult();
+  }
+
+  @Override
+  public abstract String getSiteName(String siteId);
+
+  /**
+   * Searches for a {@link DirectQuery} within this broker. The specified query ID identifies the
+   * query.
+   *
+   * @param queryId The identifier of the query.
+   * @return The query if there is any.
+   * @throws QueryNotFoundException If the ID does not identify a known query.
+   */
+  protected DirectQuery findQuery(String queryId) throws QueryNotFoundException {
+    return Optional.ofNullable(brokerQueries.get(queryId))
+        .orElseThrow(() -> new QueryNotFoundException(queryId));
+  }
+
+  /**
+   * "Obfuscates" a number by adding or subtracting a random number <=5.
+   * @param resultCount the precise result
+   * @return The obfuscated result, or 0 if the obfuscated result is < 5
+   */
+  protected int obfuscate(int resultCount) {
+    int randomNum = ThreadLocalRandom.current().nextInt(0, Integer.MAX_VALUE);
```

Comment:
Why don't you just choose integers between 0 and 11 (exclusive) and don't use modulo?